clients/fossid-webapp/src/main/kotlin/model/rules/IgnoreRule.kt

```diff
@@ -50,4 +50,8 @@ data class IgnoreRule(
      * The date the rule was last updated (e.g. 2022-01-05 14:59:22).
      */
     val updated: String
-)
+
+) {
+    override fun equals(other: Any?): Boolean =
+        (other === this) || (other is IgnoreRule && other.type == type && other.value == value)
```

Comment:
Overriding `equals` this way without also overriding `hashCode` might cause all kinds of nasty issues. Can this be replaced by a custom function with a telling name about which properties are being used / ignored?

Reply:
What about using `distinctBy {}`?

Reply:
How would you use it in the `filterLegacyRules` function ?
Phrasing it differently, what the intersection of the two lists would bring me ?

Reply:
Maybe I should use `any` and do the comparison myself.

Reply:
Right, `any` makes more sense, probably like

    rulesToTest.filterNot { ruleToTest ->
        any { it.type == ruleToTest.type && it.value == ruleToTest.value }
    }